v2/pkg/protocols/common/executer/executer.go

```diff
@@ -71,6 +72,7 @@ func (e *Executer) Execute(input *contextargs.Context) (bool, error) {
 	}
 	previous := make(map[string]interface{})
 
+	mtx := &sync.Mutex{}
```

Comment:
this mutex seems unnecessary as the variable has write-only operations within two mutually exclusive code branches. Could you verify if removing it reintroduces somehow the data race?

Reply:
yes @Mzack9999, I have tried again somehow it's reintroduces data race issue without a mutex wrapper.
```
[openssh-detect] [tcp] [info] scanme.sh:22 [SSH-2.0-OpenSSH_8.2p1 Ubuntu-4ubuntu0.9]
==================
WARNING: DATA RACE
Write at 0x00c00d251da0 by goroutine 7295350:
  github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(*Executer).Execute.func3()
      /Users/ramana/projectdiscovery-workspace/nuclei/v2/pkg/protocols/common/executer/executer.go:
```